### PR TITLE
Re connect

### PR DESCRIPTION
**Pull request by AleksandarFilipov**
Client can now reconnect.

test scenario
https://github.com/aeternity/ae-channel-service/pull/7/files#diff-3d9f608962f0b90ad220aac134a19cdcR213

which executes relevant code here
https://github.com/aeternity/ae-channel-service/pull/7/files#diff-835341fcd3816d33d40b79db245d9db5R141

Sophia compiler update, this TicTacToe is updated. And patches executed from clone has been updated/removed

### CODE REVIEW

**Review by tolbrino — COMMENTED** [review]
What is the PR about and what should I focus in terms of reviewing it?

File: apps/ae_socket_connector/lib/socket_connector.ex [hunk 1]
```diff
@@ -149,6 +136,78 @@ defmodule SocketConnector do
...
+      :aesc_client_reconnect_tx.new(%{
+        :channel_id => :aeser_id.create(tag, channel),
+        :round => nonce,
+        :role => role,
+        :pub_key => :aeser_id.create(:account, pub_key)
+      })
+
+    # now we just sign it.
+    # TODO reuse code from singer....
+    bin = :aetx.serialize_to_binary(aetx)
```
Comment (AleksandarFilipov) [inline apps/ae_socket_connector/lib/socket_connector.ex:189]: this sign code is beeing moved to Signer.

File: apps/ae_socket_connector/lib/client_runner.ex [hunk 2]
```diff
@@ -210,8 +210,66 @@ defmodule ClientRunner do
     {jobs_initiator, jobs_responder}
   end
 
+  def reconnect_jobs() do
```
Comment (AleksandarFilipov) [inline apps/ae_socket_connector/lib/client_runner.ex:213]: Theis is the test scenario for re-connect

File: apps/ae_socket_connector/lib/socket_connector.ex [hunk 3]
```diff
@@ -149,6 +136,78 @@ defmodule SocketConnector do
     # WebSockex.start_link(ws_url, __MODULE__, %{priv_key: priv_key, pub_key: pub_key, role: role, session: state_channel_context, color: [ansi_color: color]}, name: name)
   end
 
+  # "ws://localhost:3014/channel?port=14035&protocol=json-rpc&reconnect_tx=tx_%2BJ0LAfhCuEBxaFk2dtVESM%2BzvaLXl319O%2B3%2FKYeLKk9pTEBCQsdAxR85LmHLHuI7gh7kuDE0X0iU33CymvyZhYREohGQFTIOuFX4U4ICPwGhBhcVvJBxDP091oeKLKHW8agBzefkBFITZPJHUXayn9anAYlpbml0aWF0b3KhASLZizA%2BhxzczTNVDD0TYeYxI0%2BWU4ivbUiUdyc9vIoepDdZMA%3D%3D&role=initiator"
```
Comment (AleksandarFilipov) [inline apps/ae_socket_connector/lib/socket_connector.ex:139]: reconnect code added below

**Review by tolbrino — APPROVED** [review]

**Pull request merged**